src/main/java/com/google/devtools/build/lib/query2/cquery/ConfiguredTargetQueryEnvironment.java

```diff
@@ -517,18 +517,15 @@ protected BuildConfiguration getConfiguration(ConfiguredTarget target) {
   }
 
   @Override
-  protected ConfiguredTargetKey getSkyKey(ConfiguredTarget target) {
-    return ConfiguredTargetKey.builder()
-        .setConfiguredTarget(target)
-        .setConfiguration(getConfiguration(target))
-        .build();
+  protected ConfiguredTargetKey getSkyKey(KeyedConfiguredTarget target) {
```

Comment:
I think this method's assumptions have caused other issues in the past so it's nice we can tighten it up here.